### PR TITLE
Fix documentation for mount state

### LINKED ISSUE #18948: mount documentation incorrectly specifies what "absent" and "unmounted" do
_From @maxrothman on August 5, 2015 15:19_

Issue Type: Bug Report (documentation)
Component Name: mount module

Ansible Version: 1.9.1

Summary:
Running the `mount` module with `state=unmounted` unmounts the device, but does not remove it from fstab. Running the mount module with `state=absent` both unmounts the device **and** removes it from fstab.

This conflicts with the [documentation](http://docs.ansible.com/ansible/mount_module.html):

> If `mounted` or `unmounted`, the device will be actively mounted or unmounted as needed and appropriately configured in fstab. `absent` and `present` only deal with fstab but will not affect current mounting.

The present functionality seems to be intended in `mount`'s code, specifically [here](https://github.com/ansible/ansible-modules-core/blob/devel/system/mount.py#L319) and [here](https://github.com/ansible/ansible-modules-core/blob/devel/system/mount.py#L335).

`mounted` and `present` appear to work as documented.

_Copied from original issue: ansible/ansible-modules-core#1869_

### PR DESCRIPTION
**Pull request by jarrpa**
##### SUMMARY
Fixes #18948

##### ISSUE TYPE
 - Docs Pull Request

##### COMPONENT NAME
mount module

##### ANSIBLE VERSION
```
devel branch
```

Signed-off-by: Jose A. Rivera <jarrpa@redhat.com>

### CODE REVIEW

File: lib/ansible/modules/system/mount.py [hunk 1]
```diff
@@ -58,13 +58,16 @@
...
       - If C(unmounted), the device will be unmounted without changing I(fstab).
-      - C(absent) and C(present) only deal with I(fstab) but will not affect
-        current mounting.
+      - C(present) only specifies that the device is to be configured in
+        I(fstab) and does not trigger or require a mount.
+      - C(absent) specifies that the device mount's entry will be removed from
+        I(fstab) and will also unmount the device and remove the mount
+        point.
       - If specifying C(mounted) and the mount point is not present, the mount
         point will be created.
```
Comment (dustymabe) [inline lib/ansible/modules/system/mount.py]: can delete this line now I think.

File: lib/ansible/modules/system/mount.py [hunk 2]
```diff
@@ -58,13 +58,14 @@
...
-      - If specifying C(mounted) and the mount point is not present, the mount
+        configured in I(fstab). If the mount point is not present, the mount
         point will be created.
-      - Similarly, specifying C(absent) will remove the mount point directory.
+      - If C(unmounted), the device will be unmounted without changing I(fstab).
+      - C(present) only specifies that the device is to be configured in
+        I(fstab) and does not trigger or require a mount.
+      - C(absent) specifies that the device mount's entry will be removed from
+        I(fstab) and will also unmount the device and remove the mount
+        point.
```
Comment (dustymabe) [inline lib/ansible/modules/system/mount.py:68]: should we say `will also unmount the device (if mounted) and remove the mount point`

**Comment by dustymabe** [comment]
LGTM

File: lib/ansible/modules/system/mount.py (hunk 2 above)
Comment (samdoran) [reply lib/ansible/modules/system/mount.py:68]: I think this is clear enough.

**Pull request merged**